src/tool-header.scss

```diff
@@ -0,0 +1,295 @@
+    max-width: 100%;
+    overflow: hidden;
+    height: $fd-tool-header-height;
+
+    &:not(:last-child) {
+      @include fd-set-margin-right(0.5rem);
+    }
+  }
+
+  &__logo {
+    @include fd-reset();
+
+    width: auto;
+    height: 1.625rem;
+    object-fit: cover;
+    background-size: contain;
+    background-repeat: no-repeat;
+  }
+
+  .#{$block}__text {
+    @include fd-ellipsis();
+
+    color: var(--sapShell_TextColor);
+  }
+
+  .#{$block}__label {
+    @include fd-ellipsis();
+    @include fd-reset-margins();
+
+    align-self: center;
+    color: var(--sapShell_TextColor);
+  }
+
+  .#{$block}__title {
+    @include fd-ellipsis();
+    @include fd-set-paddings-x-equal(0.625rem);
+
+    font-weight: bold;
+    color: var(--sapShell_TextColor);
+  }
+
+  .#{$block}__button {
+    color: var(--sapShell_InteractiveTextColor);
+
+    &:hover {
+      border-color: transparent;
+      background: var(--sapShell_Hover_Background);
+    }
+
+    &:focus {
+      outline-color: var(--sapContent_ContrastFocusColor);
+
+      &::after {
+        border-color: var(--sapContent_ContrastFocusColor);
+      }
+    }
+
+    &:active {
+      color: var(--sapShell_Active_TextColor);
+      background: var(--sapShell_Active_Background);
+      outline-color: var(--sapContent_ContrastFocusColor);
+
+      &::after {
+        border-color: var(--sapContent_ContrastFocusColor);
+      }
+    }
+
+    &-text {
+      color: var(--sapShell_TextColor);
+
+      @media (max-width: $fd-tool-header-screen-size-sm) {
+        display: none;
+      }
+    }
+
+    .#{$block}__title {
+      @include fd-reset-paddings-x();
+
+      @media (max-width: $fd-tool-header-screen-size-sm) {
+        display: none;
+      }
+    }
+  }
+
+  .#{$block}__icon {
+    color: var(--fdTool_Header_Icon_Color_Default);
+    font-size: 1rem;
+    line-height: 1;
+
+    &--inverted {
+      color: var(--fdTool_Header_Icon_Color_Inverted_Default);
+      background: var(--fdTool_Header_Icon_Background_Color_Inverted_Default);
+    }
+
+    @each $set-name, $set-params in $fd-tool-header-icon-semantic-values {
+      &--#{$set-name} {
+        color: map_get($set-params, 'iconColor');
+
+        &.#{$block}__icon--inverted {
+          color: map_get($set-params, 'iconColorInverted');
+          background: map_get($set-params, 'bgColorInverted');
+        }
+      }
+    }
+  }
+
+  .#{$block}__object-status {
+    @each $set-name, $set-params in $fd-tool-header-object-status-semantic-values {
+      &--#{$set-name} {
+        color: map_get($set-params, 'color');
+        text-shadow: none;
+
+        [class*="sap-icon"] {
+          color: map_get($set-params, 'color');
+        }
+
+        &[class*="inverted"] {
+          text-shadow: none;
+          color: map_get($set-params, 'iconColorInverted');
+
+          [class*="icon"] {
+            color: map_get($set-params, 'iconColorInverted');
+          }
+
+          background: map_get($set-params, 'bgColorInverted');
```

Comment:
Maybe move this block into mixin?

Reply:
Mixins can not be defined within control directives or other mixins. In this case we are inside an `@each` loop and rely on variables like `$set-params` which can't be accessed outside the loop.